### PR TITLE
Migration to serde-json.

### PR DESCRIPTION
**Pull request by humb1t**
Signed-off-by: Nikita Puzankov <humb1t@yandex.ru>

### CODE REVIEW

File: iroha/src/crypto.rs [hunk 1]
```diff
@@ -28,7 +29,9 @@ pub struct KeyPair {
 }
 
 /// Public Key used in signatures.
-#[derive(Encode, Decode, Ord, PartialEq, Eq, PartialOrd, Debug, Clone, Hash, Default)]
+#[derive(
+    Copy, Encode, Decode, Ord, PartialEq, Eq, PartialOrd, Debug, Clone, Hash, Default, Deserialize,
+)]
 pub struct PublicKey {
     inner: [u8; 32],
```
Comment (vmarkushin) [inline iroha/src/crypto.rs:36]: I think serialized data will look more pretty if we will use `#[serde(flatten)]` here
```suggestion
    #[serde(flatten)]
    inner: [u8; 32],
```

File: iroha/src/crypto.rs [hunk 2]
```diff
@@ -28,7 +29,9 @@ pub struct KeyPair {
 }
 
 /// Public Key used in signatures.
-#[derive(Encode, Decode, Ord, PartialEq, Eq, PartialOrd, Debug, Clone, Hash, Default)]
+#[derive(
+    Copy, Encode, Decode, Ord, PartialEq, Eq, PartialOrd, Debug, Clone, Hash, Default, Deserialize,
```
Comment (vmarkushin) [inline iroha/src/crypto.rs:33]: Thanks for the `Clone`! :)

File: iroha/src/crypto.rs [hunk 3]
```diff
@@ -52,43 +55,27 @@ impl TryFrom<Vec<u8>> for PublicKey {
...
+#[derive(Clone, Debug, Deserialize, PartialEq)]
 pub struct PrivateKey {
-    inner: [u8; 64],
-}
-
-impl Debug for PrivateKey {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        fmt::Debug::fmt(&&self.inner[..], f)
-    }
+    inner: Vec<u8>,
```
Comment (vmarkushin) [inline iroha/src/crypto.rs:60]: The same suggestion here...
```suggestion
    #[serde(flatten)]
    inner: Vec<u8>,
```

File: iroha/src/sumeragi.rs [hunk 4]
```diff
@@ -64,7 +64,7 @@ impl Sumeragi {
                 config.max_faulty_peers,
             )
             .init()?,
-            peer_id: config.peer_id,
+            peer_id: PeerId::new(&config.torii_url, &config.public_key),
```
Comment (eadventurous) [inline iroha/src/sumeragi.rs:67]: Maybe provide `PeerId::from(config: ... )` or something similar.

File: iroha/tests/test_config.json [hunk 5]
```diff
@@ -1,9 +1,22 @@
...
+      "public_key": {"inner": [101, 170, 80, 164, 103, 38, 73, 61, 223, 133, 83, 139, 247, 77, 176, 84, 117, 15, 22, 28, 155, 125,     80, 226, 40, 26, 61, 248, 40, 159, 58, 53]}
+    },
+    {
+      "address": "localhost:1338",
+      "public_key": {"inner": [101, 170, 80, 164, 103, 38, 73, 61, 223, 133, 83, 139, 247, 77, 176, 84, 117,     15, 22, 28, 155, 125, 80, 226, 40, 26, 61, 248, 40, 159, 58, 53]}
+    },
+    {
+      "address": "195.162.0.1:23",
+      "public_key": {"inner": [101, 170, 80, 164, 103, 38, 73, 61, 223, 133, 83, 139, 247, 77, 176, 84, 117, 15, 22, 28, 155, 125, 80, 226, 40, 26, 61, 248, 40, 159, 58, 53]}
+    }
```
Comment (eadventurous) [inline iroha/tests/test_config.json:18]: Let's move trusted peers to a separate file as I think we agreed. Now with `serde_json` it should be simple. Or you want to do it in a separate PR?

File: iroha/src/crypto.rs (hunk 1 above)
Comment (humb1t) [reply iroha/src/crypto.rs:36]: 1. we will replace serialization with hex
2. We do not have serialization to json right now

File: iroha/tests/test_config.json (hunk 5 above)
Comment (humb1t) [reply iroha/tests/test_config.json:18]: This and previous comment will be addressed in the PR with `config` decomposition

File: iroha/src/crypto.rs (hunk 1 above)
Comment (vmarkushin) [reply iroha/src/crypto.rs:36]: @humb1t yeah, but we have deserialization from JSON config

Comment (humb1t) [reply iroha/src/crypto.rs:36]: @vmarkushin yes - I understand, I thought that after https://jira.hyperledger.org/browse/IR-745 we will have two fields (inner + type) but hex representation will combine both fields inside one - so I will apply `flatten`. Thanks one more time.

File: iroha/src/crypto.rs (hunk 3 above)
Comment (humb1t) [reply iroha/src/crypto.rs:60]: `Error("can only flatten structs and maps", line: 1, column: 197)`
Looks like we can't flatten Vec - I assume we will need to work on it in https://jira.hyperledger.org/browse/IR-745

Comment (vmarkushin) [reply iroha/src/crypto.rs:60]: @humb1t oh, ok. There is also `#[serde(transparent)]` for a struct itself. Can you try it?

Comment (humb1t) [reply iroha/src/crypto.rs:60]: @vmarkushin I will try to play with it in IR-745
